hdf/examples/AN_get_annotation_info.c

```diff
@@ -1,5 +1,11 @@
 #include "hdf.h"
 
+/* Used to make certain a return value _is_not_ a value.  If not ture, */
+/* print error messages, increment num_err and return. */
+#define CHECK(ret, val, where) \
+do {if(ret == val) {printf("*** ERROR from %s is %ld at line %4d in %s\n", where, (long)ret, (int)__LINE__,__FILE__);} \
+} while(0)
+
```

Comment:
A header file should be added and this macro should go in there instead of every .c file.  On the other hand, that structure could be an issue for the examples to be posted on the portal pages... Could it?  I looked in hdf5 to see how it was done.  The returned values are not checked.  I think years ago, we agreed on that for examples...

Reply:
That was the first idea, but there was no header file, second best add the macro.
One more option - add the macro to a public header that does get included.

Reply:
Why can't we just add a new header file for the examples, like in the test directory?  Will that cause complication that I don't know?

Reply:
Another header then has to be managed - does it get published (handled by make install) , is it safe and documented for use by users, are there dependencies. Should we include other useful macros. Maybe create a test macros header, that just included by another header. All this for a very legacy product?

Reply:
I see your points, but I don't think it will be complicated because it will be local to the examples directory only.  Managing it is as simple as managing hdftest.h in the test directory.  Also, you might run into the duplicate definition of CHECK issue if the examples interact with each other somehow in the future.  It's very unlikely but not impossible.  Regarding your concerns:
- does it get published (handled by make install): yes, for users to use it too
- is it safe and documented for use by users: yes, it will be a very simple header file
- are there dependencies: no reasons to be
- Should we include other useful macros: definitely for these examples, that's the essential benefit of its existance

Sorry, you lost me here: Maybe create a test macros header, that just included by another header.

Another significant and much likely benefit is avoiding an error-prone situation.  If you want to change something in the macros, only one version will be changed, instead of many source files, as you already saw for correcting "If not ture,".

I'm willing to do the work if that will help.

Reply:
I agree, one file to include is best.
It's a small useful macro - just add it to an appropriate public header that already is installed.

Reply:
OK.  At the end of hdf.h is good since both sd and hdf examples can have it.  Can you reverse the addition of the macro definition to the source files?  I'll add it after you merge.  If it's too much hassle to reverse, I can manually remove them.